# HG changeset patch
# User andrew
# Date 1511841133 0
#      Tue Nov 28 03:52:13 2017 +0000
# Node ID bccc53c13c22acbc87a3829830ffb736d8c18bab
# Parent  24d492352bf29b67576fab7d46d86e6c698498f9
PR3497: AArch64: Adapt to 8002074: Support for AES on SPARC

diff --git a/src/cpu/aarch64/vm/aarch64.ad b/src/cpu/aarch64/vm/aarch64.ad
--- openjdk/hotspot/src/cpu/aarch64/vm/aarch64.ad
+++ openjdk/hotspot/src/cpu/aarch64/vm/aarch64.ad
@@ -1590,6 +1590,12 @@
   return Op_RegL;
 }
 
+// AArch64 AES instructions are compatible with SunJCE expanded
+// keys, hence we do not need to pass the original key to stubs
+const bool Matcher::pass_original_key_for_aes() {
+  return false;
+}
+
 // x86 supports misaligned vectors store/load.
 const bool Matcher::misaligned_vectors_ok() {
   // TODO fixme
